implementation/src/main/java/io/smallrye/jwt/config/JWTAuthContextInfoProvider.java

```diff
@@ -183,11 +213,29 @@ private static JWTAuthContextInfoProvider create(String publicKey, String public
             contextInfo.setTokenHeader(tokenHeader);
         }
         SmallryeJwtUtils.setContextTokenCookie(contextInfo, tokenCookie);
+        if (defaultSubjectClaim != null && defaultSubjectClaim.isPresent()) {
+            contextInfo.setDefaultSubjectClaim(defaultSubjectClaim.get());
+        }
         if (defaultGroupsClaim != null && defaultGroupsClaim.isPresent()) {
             contextInfo.setDefaultGroupsClaim(defaultGroupsClaim.get());
         }
         SmallryeJwtUtils.setContextGroupsPath(contextInfo, groupsPath);
 
+        if (expGracePeriodSecs != null && expGracePeriodSecs.isPresent()) {
+            contextInfo.setExpGracePeriodSecs(expGracePeriodSecs.get());
+        }
+
+        if (whitelistAlgorithms != null && !whitelistAlgorithms.isEmpty()) {
+            contextInfo.getWhitelistAlgorithms().addAll(whitelistAlgorithms);
+        }
+
+        // JWKS based on jose4j
+        if (jwksLocation != null && jwksLocation.isPresent()) {
+            contextInfo.setJwksUri(jwksLocation.get());
+            contextInfo.setJwksRefreshInterval(jwksRefreshInterval);
+            contextInfo.setFollowMpJwt11Rules(false);
```

Comment:
The more I'm looking at this the more I'm uncomfortable (not with your code :-) ) but with the way it is currently done in smalrye-jwt (if jwks happens to be on the classpath then they are loaded from one resolver, if it is https uri - then it is done inside the parser) and then this follow the rules property which is confusing as well.
Please drop this code for now, I will open an issue to consolidate (I'd rather do it then making this separation more visible which would be the case with the new properties you propose) and will link to it below. Keep the `jwksRefreshInterval` property though please, we can update the docs later to say it is ignored if jwks are not on the classpath.

Reply:
I see. Yes, I understand your point :)

The`jwksRefreshInterval` is only being applied in the http KeyResolver and not in the class path resolver, which may be more confusing. That was the reason why I wanted to use the http KeyResolver.

Maybe we could use the jose4j HttpsJwks and implement our own SimpleGet to also support loading JWKS from the classpath, and basically drop the current code used to load JWKS and use the jose4j one.

Reply:
https://github.com/smallrye/smallrye-jwt/issues/82.

Let me summarize all below, thanks